cmd/common/utils/flags.go

```diff
@@ -152,3 +153,41 @@ func AddRegistryAuthVariablesAndFlags(cmd *cobra.Command, authOptions *oci.AuthO
 		"Allow connections to HTTP only registries",
 	)
 }
+
+// RemoveSplitSortArgs removes the --sort flag with its arg, if it isn't in the
+// merged form of --sort=foo
+func RemoveSplitSortArgs(args []string) []string {
+	for i := 0; i < len(args); i++ {
+		if args[i] == "--sort" {
+			// Remove also the next element as it is the arg of --sort
+			return append(args[:i], args[i+2:]...)
+		}
+	}
+	return args
+}
+
+// RemoveHelpArg removes the --help flag
+func RemoveHelpArg(args []string) []string {
+	for i := 0; i < len(args); i++ {
+		if args[i] == "--help" || args[i] == "-h" {
+			return append(args[:i], args[i+1:]...)
+		}
+	}
+	return args
+}
+
+func ParseEarlyFlags(cmd *cobra.Command) error {
+	// Do not error out on unknown flags, but still validate currently
+	// known ones.
+	// Other flags will be validated in the `Execute()` call and unknown
+	// ones will be rejected
+
+	cmd.FParseErrWhitelist.UnknownFlags = true
```

Comment:
nit: Looks like this places where a nice `defer` would make code clearer (even if it doesn't provide any value 🙂):

```suggestion
	cmd.FParseErrWhitelist.UnknownFlags = true
	defer func() {
		cmd.FParseErrWhitelist.UnknownFlags = false
	}()
```

Reply:
I am always a fan of defer. Thanks